File: spec/java/src/io/kaitai/struct/spec/TestMultipleUse.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.MultipleUse;
import org.testng.annotations.Test;

import static org.testng.Assert.assertEquals;

public class TestMultipleUse extends CommonSpec {
    @Test
    public void testMultipleUse() throws Exception {
        MultipleUse r = MultipleUse.fromFile(SRC_DIR + "position_abs.bin");

        assertEquals(r.t1().firstUse().value(), 0x20);
        assertEquals(r.t2().secondUse().value(), 0x20);
    }
}

File: spec/javascript/test_multiple_use.js
var assert = require('assert');
var testHelper = require('testHelper');

testHelper('MultipleUse', 'src/position_abs.bin', function(r) {
  assert.equal(r.t1.firstUse.value, 0x20);
  assert.equal(r.t2.secondUse.value, 0x20);
});
